test/rpc-test.js

```diff
@@ -0,0 +1,176 @@
+/* eslint-env mocha */
+/* eslint prefer-arrow-callback: "off" */
+
+'use strict';
+
+const assert = require('./util/assert');
+const consensus = require('../lib/protocol/consensus');
+const FullNode = require('../lib/node/fullnode');
+const Network = require('../lib/protocol/network');
+const network = Network.get('regtest');
+
+const node = new FullNode({
+  network: 'regtest',
+  apiKey: 'foo',
+  walletAuth: true,
+  memory: true,
+  workers: true,
+  plugins: [require('../lib/wallet/plugin')]
+});
+
+const {NodeClient, WalletClient} = require('bclient');
+
+const nclient = new NodeClient({
+  port: network.rpcPort,
+  apiKey: 'foo'
+});
+
+const wclient = new WalletClient({
+  port: network.walletPort,
+  apiKey: 'foo'
+});
+
+const {wdb} = node.require('walletdb');
+
+let addressHot = null;
+let addressMiner = null;
+let walletHot = null;
+let walletMiner = null;
+let blocks = null;
+let txid = null;
+let utxo = null;
+
+describe('RPC', function() {
+  this.timeout(15000);
+
+  it('should open node and create wallets', async () => {
+    consensus.COINBASE_MATURITY = 0;
+    await node.open();
+    await nclient.open();
+    await wclient.open();
+
+    const walletHotInfo = await wclient.createWallet('hot');
+    walletHot = wclient.wallet('hot', walletHotInfo.token);
+    const walletMinerInfo = await wclient.createWallet('miner');
+    walletMiner = wclient.wallet('miner', walletMinerInfo.token);
+    await walletHot.open();
+    await walletMiner.open();
+  });
+
+  it('should rpc help', async () => {
+    assert( await nclient.execute('help', []) );
+    assert( await wclient.execute('help', []) );
+
+    await assert.rejects( async () => await nclient.execute('help',
```

Comment:
`assert.rejects` is available with Node.js v10+, and not defined in v8 LTS.

Reply:
Yeah I know. @tuxcanfly told me on slack that CI at least runs tests on node V10. If this is an issue for users or backwards compatibility I'll need a different way to test that a promise got rejected. (`assert.throws` will not work for async errors apparently)

Reply:
We should probably run CI with the current LTS and the "next" release with latest features.

Reply:
How about https://github.com/braydonf/bcoin/commit/37289786117a9f0ee340936345a07657ed1837c2 ?

Reply:
Just pushed commit that adds a util function for testing errors thrown by async functions. Works in V8 and V10, positive and negative test works correctly.